packages/doenetml-worker-rust/lib-doenetml-core/src/core/document_model/dependency_creation_helpers/data_query_pick_prop.rs

```diff
@@ -45,3 +47,84 @@ pub fn pick_prop(
 
     None
 }
+
+/// For each node in `nodes`, picks props that match the profiles of `match_profiles_list`,
+/// where the number of props picked per node is determined by the length of `match_profiles_list`.
+/// If `limit_to_first_match` is true, props are picked only from the first node with a match;
+/// otherwise props are picked from all nodes that match.
+/// Returns a vector of the edges needed to go from `query_node` to the props.
+///
+/// If more then one prop is being picked per node, then virtual nodes will be created and added to dependency graph.
+/// One virtual node will be added to represent each node from `nodes` with matching props.
+/// If a subset of the props are matched for a given node, virtual nodes will be added as placeholders for missing props.
+pub fn process_pick_prop<N>(
+    nodes: N,
+    match_profiles_list: Vec<Vec<PropProfile>>,
+    limit_to_first_match: bool,
+    query_node: GraphNode,
+    document_structure: &DocumentStructure,
+    document_model: &DocumentModel,
+) -> Vec<(GraphNode, GraphNode)>
+where
+    N: Iterator<Item = GraphNode>,
+{
+    if match_profiles_list.len() == 1 {
```

Comment:
Checking against len `1` and not len `0` makes me nervous.

Reply:
It does do the right thing and return an empty vector if the length is 0 (given that `.any()` returns false an empty iterator). It's not an efficient way to get to the empty vector, but it's not a case that should occur.